File: applications/luci-app-nut/luasrc/model/cbi/nut_cgi.lua
-- Copyright 2015 Daniel Dickinson <openwrt@daniel.thecshore.com>
-- Licensed to the public under the Apache License 2.0.

local m, s, o

m = Map("nut_cgi", translate("Network UPS Tools (CGI)"),
	translate("Network UPS Tools CGI Configuration"))

s = m:section(DummySection)

function s.set(self, section, option, value)
end

function s.del(self, section)
end

function s.remove(self, section)
end

o=s:option(SimpleButton, "nut_cgi_page", translate('Go to NUT CGI Page'))
o.section = { }
o.section[".name"] = "1"

function o.write(self, section, value)
	luci.http.redirect("/nut")
end

s = m:section(TypedSection, "host", translate("Host"))
s.addremove = true
s.anonymous = true

o = s:option(Value, "upsname", translate("UPS name"), translate("As configured by NUT"))
o.optional = false

o = s:option(Value, "hostname", translate("Hostname or IP address"))
o.optional = false
o.datatype = "host"

o = s:option(Value, "port", translate("Port"))
o.datatype = "port"
o.optional = true
o.placeholder = 3493

o = s:option(Value, "displayname", translate("Display name"))
o.optional = false

return m

File: applications/luci-app-nut/htdocs/luci-static/resources/view/nut/nut_cgi.js
'use strict';
'require form';
'require view';

return view.extend({
	load: function() {

	},

	render: function() {
		let m, s, o;

		m = new form.Map('nut_cgi', _('NUT CGI'),
			_('Network UPS Tools CGI Configuration') + '<br />' +
			'%s'.format('<a href="/nut">%s</a>'.format(_('Go to NUT CGI'))));

		s = m.section(form.TypedSection, 'host', _('Host'));
		s.addremove = true;
		s.anonymous = true;

		o = s.option(form.Value, 'upsname', _('UPS name'), _('As configured by NUT'));
		o.optional = false;

		o = s.option(form.Value, 'hostname', _('Hostname or IP address'));
		o.optional = false;
		o.datatype = 'or(host,ipaddr)';

		o = s.option(form.Value, 'port', _('Port'));
		o.datatype = 'port';
		o.optional = true;
		o.placeholder = 3493;

		o = s.option(form.Value, 'displayname', _('Display name'));
		o.optional = false;

		s = m.section(form.TypedSection, 'upsset', _('Control UPS via CGI'));
		s.addremove = false;
		s.anonymous = true;
		s.optional = false;

		o = s.option(form.Flag, 'enable', _('Enable'));
		o.optional = false;
		o.default = false;

		return m.render();
	}
});
